src/btreemap/allocator.rs

```diff
@@ -45,7 +45,7 @@ pub struct Allocator<M: Memory> {
     memory: M,
 }
 
-#[repr(packed)]
+#[repr(C, packed)]
```

Comment:
Maybe we can keep this change (and other similar changes) out of this PR as it isn't part of updating the `BTreeHeader` implementation? This change will also be obsolete once we migrate these to use the same pattern as `BTreeHeader`, correct?

Reply:
This PR is just built on top of https://github.com/dfinity/stable-structures/pull/38 The idea was to merge `#[repr(C, packed)]` ASAP, and then agree on the unaligned pointers in this MR...